extensions/azurecore/src/azureResource/providers/postgresArcServer/postgresServerProvider.ts

```diff
@@ -0,0 +1,28 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ExtensionContext } from 'vscode';
+import { ApiWrapper } from '../../../apiWrapper';
+
+import { azureResource } from '../../azure-resource';
+import { IAzureResourceService } from '../../interfaces';
+import { PostgresServerArcTreeDataProvider as PostgresServerArcTreeDataProvider } from './postgresServerTreeDataProvider';
+
+export class PostgresServerArcProvider implements azureResource.IAzureResourceProvider {
+	public constructor(
+		private _databaseServerService: IAzureResourceService<azureResource.AzureResourceDatabaseServer>,
+		private _apiWrapper: ApiWrapper,
+		private _extensionContext: ExtensionContext
+	) {
+	}
+
+	public getTreeDataProvider(): azureResource.IAzureResourceTreeDataProvider {
+		return new PostgresServerArcTreeDataProvider(this._databaseServerService, this._apiWrapper, this._extensionContext);
+	}
+
+	public get providerId(): string {
+		return 'azure.resource.providers.postgresArcServer';
```

Comment:
If users don't have the postgres extension installed, we should leverage the recommendation service to suggest installing it when we have more management capabilities.

Reply:
Good idea - let's sync up offline for how we'd want to handle this.

Reply:
This is something we would do in core and so is probably outside the scope of this PR